### PR TITLE
Details about the computation of the pdf

### PR DESCRIPTION
**Pull request by ThoChat**
Can you please check if the new details make sens for you and if you see any spelling mistakes.
Cheers!

### CODE REVIEW

**Comment by mirakuepper** [comment]
I have one additional question about randomization: Are the future trajectory points of a given person taken into account as well or only the position of all other people?

**Comment by ThoChat** [comment]
> I have one additional question about randomization: Are the future trajectory points of a given person taken into account as well or only the position of all other people?

I'll not sure I understood the question correctly. Are you talking about the randomization process or about the computation of $P_{Ni}$?

**Comment by mirakuepper** [comment]
Ok, I'll try to explain my question:
For P(r) we calculate the distance each pedestrian has to all other people in the same frame. This we do for all persons and all frames and then we get the probability P(r).

But what do we do for PNi? 
Do we ignore the frames and then see how the distances for all people are distributed? 
Or do we drop the frame-information and then pick positions at random (but how many do we pick?)? 
Or do we calculate this per person as well? Like taking one persons position and then calculate the distance to the randomised positions (but if we do this, a person could be compared to his own future/past path)?

 This somehow really confuses me...

**Comment by ThoChat** [comment]
> Ok, I'll try to explain my question: For P(r) we calculate the distance each pedestrian has to all other people in the same frame. This we do for all persons and all frames and then we get the probability P(r).
> 
> But what do we do for PNi? Do we ignore the frames and then see how the distances for all people are distributed? Or do we drop the frame-information and then pick positions at random (but how many do we pick?)? Or do we calculate this per person as well? Like taking one persons position and then calculate the distance to the randomised positions (but if we do this, a person could be compared to his own future/past path)?
> 
> This somehow really confuses me...

I get it now. : )
Thanks a lot for detailing your question.
The answer is the second option you mentioned : We drop the frame-information and then pick positions at random. The number of positions picked is then also random. The random process should ensure a uniform distribution of positions for each frame, but to smooth any noise that this method may induce, it's better to use a higher `randomisation_stacking' number for the computation of the function. This is shown in the "Parameters of the PDF" section in the User guide.

**Comment by mirakuepper** [comment]
Thanks a lot! Can you maybe add the information that the number of picked positions is also randomised to the guide?  From "randomisation-stacking" I thought we did the random picking several times but not that we change the number of positions picked. 
But I think now I understand what is done here :D

File: notebooks/user_guide.ipynb [hunk 1]
```diff
@@ -3553,7 +3553,9 @@
...
     "\n",
-    "This probability is approximated here by randomising the time information (randomisation of the time frames).\n",
+    "In this method, \"interacting pedestrians\" are defined as pedestrians that are present in the same spatial domain at the same time. One should also keep in mind that in its current implementation, the method does not take into account walls and corners, which should in theory block any \"interaction\" between pedestrians on opposite sides of the obstacles.\n",
+    "\n",
+    "The probability $P_{Ni}$ is approximated here by time randomising the original trajectory file. For this randomisation process, only the frame numbers of the trajectory file are shuffled. The created \"randomised trajectories\" contains random pedestrian positions, composed only of positions which were present in the original trajectory file. This method helps to take into account preferred route choices of participants, which can be due to terrain features or social behaviours.\n",
```
Comment (chraibi) [inline notebooks/user_guide.ipynb:3558]: I don't understand the last sentence and I' m not sure it it is necessary to talk about routing in this context.

**Review by chraibi — APPROVED** [review]

**Pull request merged**